File: runtime/dart/ui/semanticsAction.ts
import { IIterable } from "../core/iterable";
import { List } from "../core/list";
import { Map } from "../core/map";
import { IMapEntry, MapEntry } from "../core/mapEntry";
declare const dart: {
    ui: {
        semanticsActionTap: (this: void) => ISemanticsAction;
        semanticsActionLongPress: (this: void) => ISemanticsAction;
        semanticsActionScrollLeft: (this: void) => ISemanticsAction;
        semanticsActionScrollRight: (this: void) => ISemanticsAction;
        semanticsActionScrollUp: (this: void) => ISemanticsAction;
        semanticsActionScrollDown: (this: void) => ISemanticsAction;
        semanticsActionIncrease: (this: void) => ISemanticsAction;
        semanticsActionDecrease: (this: void) => ISemanticsAction;
        semanticsActionShowOnScreen: (this: void) => ISemanticsAction;
        semanticsActionMoveCursorForwardByCharacter: (
            this: void
        ) => ISemanticsAction;
        semanticsActionMoveCursorBackwardByCharacter: (
            this: void
        ) => ISemanticsAction;
        semanticsActionSetSelection: (this: void) => ISemanticsAction;
        semanticsActionCopy: (this: void) => ISemanticsAction;
        semanticsActionCut: (this: void) => ISemanticsAction;
        semanticsActionPaste: (this: void) => ISemanticsAction;
        semanticsActionDidGainAccessibilityFocus: (
            this: void
        ) => ISemanticsAction;
        semanticsActionDidLoseAccessibilityFocus: (
            this: void
        ) => ISemanticsAction;
        semanticsActionCustomAction: (this: void) => ISemanticsAction;
        semanticsActionDismiss: (this: void) => ISemanticsAction;
        semanticsActionMoveCursorForwardByWord: (
            this: void
        ) => ISemanticsAction;
        semanticsActionMoveCursorBackwardByWord: (
            this: void
        ) => ISemanticsAction;
    };
};
export interface ISemanticsAction {
    index: number;
    toString: () => string;
}
export class SemanticsAction {
    public static _kTapIndex = 1 << 0;
    public static _kLongPressIndex = 1 << 1;
    public static _kScrollLeftIndex = 1 << 2;
    public static _kScrollRightIndex = 1 << 3;
    public static _kScrollUpIndex = 1 << 4;
    public static _kScrollDownIndex = 1 << 5;
    public static _kIncreaseIndex = 1 << 6;
    public static _kDecreaseIndex = 1 << 7;
    public static _kShowOnScreenIndex = 1 << 8;
    public static _kMoveCursorForwardByCharacterIndex = 1 << 9;
    public static _kMoveCursorBackwardByCharacterIndex = 1 << 10;
    public static _kSetSelectionIndex = 1 << 11;
    public static _kCopyIndex = 1 << 12;
    public static _kCutIndex = 1 << 13;
    public static _kPasteIndex = 1 << 14;
    public static _kDidGainAccessibilityFocusIndex = 1 << 15;
    public static _kDidLoseAccessibilityFocusIndex = 1 << 16;
    public static _kCustomAction = 1 << 17;
    public static _kDismissIndex = 1 << 18;
    public static _kMoveCursorForwardByWordIndex = 1 << 19;
    public static _kMoveCursorBackwardByWordIndex = 1 << 20;
    public static tap = dart.ui.semanticsActionTap();
    public static longPress = dart.ui.semanticsActionLongPress();
    public static scrollLeft = dart.ui.semanticsActionScrollLeft();
    public static scrollRight = dart.ui.semanticsActionScrollRight();
    public static scrollUp = dart.ui.semanticsActionScrollUp();
    public static scrollDown = dart.ui.semanticsActionScrollDown();
    public static increase = dart.ui.semanticsActionIncrease();
    public static decrease = dart.ui.semanticsActionDecrease();
    public static showOnScreen = dart.ui.semanticsActionShowOnScreen();
    public static moveCursorForwardByCharacter =
        dart.ui.semanticsActionMoveCursorForwardByCharacter();
    public static moveCursorBackwardByCharacter =
        dart.ui.semanticsActionMoveCursorBackwardByCharacter();
    public static setSelection = dart.ui.semanticsActionSetSelection();
    public static copy = dart.ui.semanticsActionCopy();
    public static cut = dart.ui.semanticsActionCut();
    public static paste = dart.ui.semanticsActionPaste();
    public static didGainAccessibilityFocus =
        dart.ui.semanticsActionDidGainAccessibilityFocus();
    public static didLoseAccessibilityFocus =
        dart.ui.semanticsActionDidLoseAccessibilityFocus();
    public static customAction = dart.ui.semanticsActionCustomAction();
    public static dismiss = dart.ui.semanticsActionDismiss();
    public static moveCursorForwardByWord =
        dart.ui.semanticsActionMoveCursorForwardByWord();
    public static moveCursorBackwardByWord =
        dart.ui.semanticsActionMoveCursorBackwardByWord();
    public static values = Map.fromEntries(
        <IIterable<IMapEntry<number, ISemanticsAction>>>(
            (<unknown>(
                List.fromArray([
                    new MapEntry(
                        SemanticsAction._kTapIndex,
                        SemanticsAction.tap
                    ),
                    new MapEntry(
                        SemanticsAction._kLongPressIndex,
                        SemanticsAction.longPress
                    ),
                    new MapEntry(
                        SemanticsAction._kScrollLeftIndex,
                        SemanticsAction.scrollLeft
                    ),
                    new MapEntry(
                        SemanticsAction._kScrollRightIndex,
                        SemanticsAction.scrollRight
                    ),
                    new MapEntry(
                        SemanticsAction._kScrollUpIndex,
                        SemanticsAction.scrollUp
                    ),
                    new MapEntry(
                        SemanticsAction._kScrollDownIndex,
                        SemanticsAction.scrollDown
                    ),
                    new MapEntry(
                        SemanticsAction._kIncreaseIndex,
                        SemanticsAction.increase
                    ),
                    new MapEntry(
                        SemanticsAction._kDecreaseIndex,
                        SemanticsAction.decrease
                    ),
                    new MapEntry(
                        SemanticsAction._kShowOnScreenIndex,
                        SemanticsAction.showOnScreen
                    ),
                    new MapEntry(
                        SemanticsAction._kMoveCursorForwardByCharacterIndex,
                        SemanticsAction.moveCursorForwardByCharacter
                    ),
                    new MapEntry(
                        SemanticsAction._kMoveCursorBackwardByCharacterIndex,
                        SemanticsAction.moveCursorBackwardByCharacter
                    ),
                    new MapEntry(
                        SemanticsAction._kSetSelectionIndex,
                        SemanticsAction.setSelection
                    ),
                    new MapEntry(
                        SemanticsAction._kCopyIndex,
                        SemanticsAction.copy
                    ),
                    new MapEntry(
                        SemanticsAction._kCutIndex,
                        SemanticsAction.cut
                    ),
                    new MapEntry(
                        SemanticsAction._kPasteIndex,
                        SemanticsAction.paste
                    ),
                    new MapEntry(
                        SemanticsAction._kDidGainAccessibilityFocusIndex,
                        SemanticsAction.didGainAccessibilityFocus
                    ),
                    new MapEntry(
                        SemanticsAction._kDidLoseAccessibilityFocusIndex,
                        SemanticsAction.didLoseAccessibilityFocus
                    ),
                    new MapEntry(
                        SemanticsAction._kCustomAction,
                        SemanticsAction.customAction
                    ),
                    new MapEntry(
                        SemanticsAction._kDismissIndex,
                        SemanticsAction.dismiss
                    ),
                    new MapEntry(
                        SemanticsAction._kMoveCursorForwardByWordIndex,
                        SemanticsAction.moveCursorForwardByWord
                    ),
                    new MapEntry(
                        SemanticsAction._kMoveCursorBackwardByWordIndex,
                        SemanticsAction.moveCursorBackwardByWord
                    ),
                ])
            ))
        )
    );
    public readonly index: number = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    public toString(): string {
        return this._dart_toString();
    }
}

File: lib/cfr/builtins/libs/dart/ui/semanticsAction.dart
import 'dart:core';
import 'dart:ui';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedSemanticsAction extends VMManagedBox<SemanticsAction> {
  VMManagedSemanticsAction(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['index'] = vmObject.index;
    table['toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final SemanticsAction vmObject;
}

void loadSemanticsAction(
    {required HydroState hydroState, required HydroTable table}) {
  table['semanticsActionTap'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.tap,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionLongPress'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.longPress,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionScrollLeft'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.scrollLeft,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionScrollRight'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.scrollRight,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionScrollUp'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.scrollUp,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionScrollDown'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.scrollDown,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionIncrease'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.increase,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionDecrease'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.decrease,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionShowOnScreen'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.showOnScreen,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionMoveCursorForwardByCharacter'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.moveCursorForwardByCharacter,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionMoveCursorBackwardByCharacter'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.moveCursorBackwardByCharacter,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionSetSelection'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.setSelection,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionCopy'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.copy,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionCut'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.cut,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionPaste'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.paste,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionDidGainAccessibilityFocus'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.didGainAccessibilityFocus,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionDidLoseAccessibilityFocus'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.didLoseAccessibilityFocus,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionCustomAction'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.customAction,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionDismiss'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.dismiss,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionMoveCursorForwardByWord'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.moveCursorForwardByWord,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  table['semanticsActionMoveCursorBackwardByWord'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<SemanticsAction>(
          object: SemanticsAction.moveCursorBackwardByWord,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  registerBoxer<SemanticsAction>(boxer: (
      {required SemanticsAction vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedSemanticsAction(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}
